### PR TITLE
HPCC-29361 Update README to reference developer docs

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
Typo in commit title

**Comment by GordonSmith** [comment]
@ghalliday - fixed typo

File: README.md [hunk 1]
```diff
@@ -65,10 +65,10 @@ ESP (Enterprise Services Platform) provides an easy to use interface to access E
...
-* [An introduction to the code generator - eclcc](https://github.com/hpcc-systems/HPCC-Platform/blob/master/devdoc/CodeGenerator.rst)
-* [The memory manager used by roxie and thor](https://github.com/hpcc-systems/HPCC-Platform/blob/master/devdoc/MemoryManager.rst)
-* [The structure of the initialization scripts](https://github.com/hpcc-systems/HPCC-Platform/blob/master/initfiles/DOCUMENTATION.rst)
-* [Outline of ecl-bundle](https://github.com/hpcc-systems/HPCC-Platform/blob/master/ecl/ecl-bundle/DOCUMENTATION.rst)
-* [The structure and some details of the cmake files](https://github.com/hpcc-systems/HPCC-Platform/blob/master/cmake_modules/DOCUMENTATION.rst)
-* [Building the documentation](https://github.com/hpcc-systems/HPCC-Platform/blob/master/docs/DOCUMENTATION.rst)
+* [An overview of workunits and the different stages in executing a query](https://hpcc-systems.github.io/HPCC-Platform/devdoc/Workunits.html)
+* [An introduction to the code generator - eclcc](https://hpcc-systems.github.io/HPCC-Platform/devdoc/CodeGenerator.html)
```
Comment (ghalliday) [inline README.md:69]: html -> md for all the examples in this file?  Or have html files been generated from the md?

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
     "docs-dev": "vitepress dev devdoc",
     "docs-build": "vitepress build devdoc",
     "docs-preview": "vitepress preview devdoc",
+    "convert": "pandoc -w markdown -o ./docs/DOCUMENTATION.md ./docs/DOCUMENTATION.rst",
```
Comment (ghalliday) [inline package.json:9]: still needed?

**Review by ghalliday — COMMENTED** [review]
@GordonSmith a couple of questions

File: README.md (hunk 1 above)
Comment (GordonSmith) [reply README.md:69]: This README.md is the one that is displayed by default in GitHub, so the links need to be full to get "into" the rendered docs

File: package.json (hunk 2 above)
Comment (GordonSmith) [reply package.json:9]: Yes - so I don't have to work out how to do it next time!

**Comment by GordonSmith** [comment]
@ghalliday - bump

**Comment by GordonSmith** [comment]
@ghalliday bump

**Review by ghalliday — APPROVED** [review]

**Pull request merged**